### PR TITLE
Passport configuration and current-user query and resolver added

### PR DESCRIPTION
**Pull request by hernan-dp**

### CODE REVIEW

File: src/controllers/user-controller.js [hunk 1]
```diff
@@ -59,3 +59,7 @@ export const signIn = async (_, { data }, { models }) => {
     token
   }
 }
+
+export const userlogged = async (_, _, { req }) => {
```
Comment (abelosorio) [inline src/controllers/user-controller.js]: You aren't performing any asynchronous tasks here.

File: src/index.js [hunk 2]
```diff
@@ -5,25 +5,21 @@ import models from './models'
...
-    }
-  })
+passport.use(new Strategy({
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: process.env.JWT_SECRET
+}, async (payload, req) => {
+  try {
+    req.user = await User.findByPk(payload.sub)
+  } catch (error) {
+    throw Error(error)
```
Comment (abelosorio) [inline src/index.js]: When you catch general errors, the `error` object could be an instance of any type of error. Here, you're always converting this error to the `Error` class and, potentially, losing information. I'd suggest changing this to:
```javascript
} catch (error) {
  throw error
}
```

File: src/index.js [hunk 3]
```diff
@@ -33,9 +29,12 @@ const server = new ApolloServer({
   instrospection: true,
   playground: true,
   tracing: true,
-  context: {
-    models
-  }
+  context: ({ req }) => {
+    const user = req.user
+    if (!user) throw new Error('you must be logged in')
```
Comment (abelosorio) [inline src/index.js]: The job of `context` is to set the context, not to validate if the user is authenticated. This error should not be here. You may want to have some queries/mutations to even non-authenticated users and, with this logic, it won't allow it.

File: src/schema/resolvers/user-resolver.js [hunk 4]
```diff
@@ -3,13 +3,15 @@ import * as userController from '../../controllers/user-controller'
...
+    user: userController.findUserById,
+    currentUser: userController.userlogged 
   },
 
   Mutation: {
     createUser: userController.createUser,
     deleteUser: userController.deleteUser,
     signup: userController.signUp,
-    signin: userController.signIn
+    signin: userController.signIn,
```
Comment (abelosorio) [inline src/schema/resolvers/user-resolver.js]: No trailing commas.

File: src/schema/resolvers/user-resolver.js [hunk 5]
```diff
@@ -3,13 +3,15 @@ import * as userController from '../../controllers/user-controller'
...
+
```
Comment (abelosorio) [inline src/schema/resolvers/user-resolver.js]: No blank spaces at the end of an object.

**Review by abelosorio — CHANGES_REQUESTED** [review]

**Review by rdemasi — APPROVED** [review]

**Pull request merged**